contracts/integrations/test/internal-integration-tests/match_orders_test.ts

```diff
@@ -0,0 +1,2754 @@
+import { BlockchainBalanceStore, TokenIds } from '@0x/contracts-exchange';
+import { ReferenceFunctions as LibReferenceFunctions } from '@0x/contracts-exchange-libs';
+import { blockchainTests, constants, expect, toBaseUnitAmount } from '@0x/contracts-test-utils';
+import { ExchangeRevertErrors, orderHashUtils } from '@0x/order-utils';
+import { Order, OrderStatus, SignedOrder } from '@0x/types';
+import { BigNumber } from '@0x/utils';
+
+import { Actor, actorAddressesByName, Maker } from '../actors';
+import { DeploymentManager } from '../deployment_manager';
+import { MatchOrderTester, MatchTransferAmounts } from '../utils/match_order_tester';
+
+const { isRoundingErrorCeil, isRoundingErrorFloor } = LibReferenceFunctions;
+
+blockchainTests.resets('matchOrders integration tests', env => {
+    // The fee recipient addresses.
+    let feeRecipientLeft: Actor;
+    let feeRecipientRight: Actor;
+
+    // The address that should be responsible for matching orders.
+    let matcher: Actor;
+
+    // Market makers who have opposite maker and taker assets.
+    let makerLeft: Maker;
+    let makerRight: Maker;
+
+    // The addresses of important assets for testing.
+    let makerAssetAddressLeft: string;
+    let makerAssetAddressRight: string;
+    let feeAssetAddress: string;
+
+    let makerAssetDataLeft: string;
+    let makerAssetDataRight: string;
+    let feeAssetData: string;
+
+    let deployment: DeploymentManager;
+    let matchOrderTester: MatchOrderTester;
+    let leftId: BigNumber;
+    let rightId: BigNumber;
+
+    before(async () => {
+        deployment = await DeploymentManager.deployAsync(env, {
+            numErc20TokensToDeploy: 3,
+            numErc721TokensToDeploy: 1,
+            numErc1155TokensToDeploy: 1,
+        });
+
+        makerAssetAddressLeft = deployment.tokens.erc20[0].address;
+        makerAssetAddressRight = deployment.tokens.erc20[1].address;
+        feeAssetAddress = deployment.tokens.erc20[2].address;
```

Comment:
optional nit: I prefer assigning the token contract itself to a variable instead of the address. This cleans up the `configureERC20TokenAsync` calls a bit and the `new BlockchainBalanceStore(...)` as well